utils/utils.go

```diff
@@ -58,13 +61,51 @@ func AddLabel(obj metav1.Object, key string, value string) {
 	labels[key] = value
 }
 
-func AddAnnotation(obj metav1.Object, key string, value string) {
+func AddAnnotation(obj metav1.Object, key string, value string) bool {
 	annotations := obj.GetAnnotations()
 	if annotations == nil {
 		annotations = make(map[string]string)
 		obj.SetAnnotations(annotations)
+	} else if curr, found := annotations[key]; !found || curr != value {
+		annotations[key] = value
+		return true
 	}
-	annotations[key] = value
+	return false
+}
```

Comment:
There is a bug here. If there is no annotation map you are not adding it you are just creating the annotation map. Also, there is no reason to use `make` here  

```suggestion
func AddAnnotation(obj metav1.Object, key string, value string) bool {
	annotations := obj.GetAnnotations()
	if annotations == nil {
		annotations = {map[string]string}
		obj.SetAnnotations(annotations)
	} 
	if curr, found := annotations[key]; !found || curr != value {
		annotations[key] = value
		return true
	}
	return false
}
```